lib/src/config_parser.cc

```diff
@@ -198,6 +198,21 @@ namespace hocon { namespace config_parser {
         return make_shared<simple_config_object>(object_origin, move(values));
     }
 
+    shared_value parse_context::parse_concatenation(shared_ptr<config_node_concatenation> n) {
+        if (_flavor == config_syntax::JSON) {
+            throw config_exception("Found a concatenation node in JSON");
```

Comment:
We've implemented bug_or_broken_exception, we should try to follow the semantics of the original exception types.